### PR TITLE
Initial sourcehut dispatch file.

### PR DESCRIPTION
**Pull request by vext01**
Does only the build step for now, as I know the tests will fail.

FAO @justinmk:
> maybe you could send a PR with a build.yml for OpenBSD? Then we can see if sr.ht hook picks it up.

Fingers crossed.

P.S> Did you find my `vext01` user on sourcehut after all?

### CODE REVIEW

**Comment by justinmk** [comment]
webhook was successfully triggered! Response:

```
There are errors in openbsd.yml:
while scanning for the next token
found character '\t' that cannot start any token
  in "<byte string>", line 16, column 1:
    	export AUTOCONF_VERSION=2.69
    ^
```

**Comment by vext01** [comment]
Fixed the tabs. Is there a page I can see the build output?

**Comment by justinmk** [comment]
https://builds.sr.ht/~jmk  🎉

File: .builds/openbsd.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+image: openbsd/6.5
+
+packages:
+- gmake
+- cmake
+- libtool
+- unzip-6.0p11
+- autoconf-2.69p2
+- automake-1.15.1
```
Comment (justinmk) [inline .builds/openbsd.yml]: is `ninja` available? it will speed up the build (`gmake` will still be needed for third-party bundled Makefiles).

Comment (vext01) [reply .builds/openbsd.yml]: ninja is available, I'll add it

Comment (vext01) [reply .builds/openbsd.yml]: (IIRC the instructions for OpenBSD won't use ninja even if it's installed -- the top-level makefile is busted on OpenBSD)

**Comment by justinmk** [comment]
somehow this still shows in the log:

    /home/build/.buildenv: line 8: export: `file.': not a valid identifier

**Comment by vext01** [comment]
Has to be a sourcehut bug? My manifest doesn't contain the string `file.` at all.

File: .builds/openbsd.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- https://github.com/neovim/neovim
+
+tasks:
+- build: |
+    export AUTOCONF_VERSION=2.69
+    export AUTOMAKE_VERSION=1.15
+    cd neovim
+    mkdir .deps
+    cd .deps
+    cmake ../third-party/
```
Comment (justinmk) [inline .builds/openbsd.yml]:
```suggestion
    cmake -G Ninja ../third-party/
```

File: .builds/openbsd.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    gmake
+    cd ..
+    mkdir build
+    cd build
+    cmake ..
```
Comment (justinmk) [inline .builds/openbsd.yml]:
```suggestion
    cmake -G Ninja ..
```

File: .builds/openbsd.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    gmake
```
Comment (justinmk) [inline .builds/openbsd.yml]: this and the other `gmake` invocation should be `ninja`.

File: .builds/openbsd.yml (hunk 1 above)
Comment (justinmk) [reply .builds/openbsd.yml]: the top-level Makefile is only a convenience for those who don't want to use CMake directly, we can ignore it here.

**Comment by vext01** [comment]
Sourcehut build now succeeding.

**Comment by vext01** [comment]
Shall I squash this?

File: .builds/openbsd.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+tasks:
+- build: |
+    export AUTOCONF_VERSION=2.69
+    export AUTOMAKE_VERSION=1.15
+    cd neovim
+    mkdir .deps
+    cd .deps
+    cmake -G Ninja ../third-party/
+    ninja
```
Comment (justinmk) [inline .builds/openbsd.yml]: changed to `cmake --build . --config Debug`.

**Comment by blueyed** [comment]
Thanks for this!

> Does only the build step for now, as I know the tests will fail.

What's the problem there?

**Comment by vext01** [comment]
> What's the problem there?

When I ran them the other day, they hung. The newtests are not run routinely on OpenBSD as a) upstream hasn't had OpenBSD in CI, and b) when I port neovim for the OpenBSD packages, I only run oldtests, as we don't have all the lua and luajit support in the ports tree.

I'm going to make a PR which enables the tests next, and we can see if we can get them passing.

**Pull request closed without merging**